mws/apis/subscriptions.py

```diff
@@ -17,76 +18,145 @@ class Subscriptions(MWS):
-        raise NotImplementedError
+        if attribute_list is None:
+            raise ValueError("Attribute_list cannot be None")
+
+        data = {
+            "Action": "RegisterDestination",
+            "MarketplaceId": marketplace_id,
+            "Destination.DeliveryChannel": delivery_channel
+        }
+        data.update(utils.enumerate_keyed_param("Destination.AttributeList.member", attribute_list))
 
-    def deregister_destination(self):
+        return self.make_request(data, "POST")
+
+    def deregister_destination(self, marketplace_id, delivery_channel="SQS", attribute_list=None):
         """
         Removes an existing destination from the list of registered destinations.
 
         Docs:
         http://docs.developer.amazonservices.com/en_US/subscriptions/Subscriptions_DeregisterDestination.html
         """
-        raise NotImplementedError
+        if attribute_list is None:
+            raise ValueError("Attribute_list cannot be None")
+
+        data = {
+            "Action": "DeregisterDestination",
+            "MarketplaceId": marketplace_id,
+            "Destination.DeliveryChannel": delivery_channel
+        }
+        data.update(utils.enumerate_keyed_param("Destination.AttributeList.member", attribute_list))
 
-    def list_registered_destinations(self):
+        return self.make_request(data, "POST")
+
+    def list_registered_destinations(self, marketplace_id):
         """
         Lists all current destinations that you have registered.
 
         Docs:
         http://docs.developer.amazonservices.com/en_US/subscriptions/Subscriptions_ListRegisteredDestinations.html
         """
-        raise NotImplementedError
+        data = {"Action": "ListRegisteredDestinations",
+                "MarketplaceId": marketplace_id}
+
+        return self.make_request(data, "POST")
 
-    def send_test_notification_to_destination(self):
+    def send_test_notification_to_destination(self, marketplace_id, delivery_channel="SQS", attribute_list=None):
         """
         Sends a test notification to an existing destination.
 
         Docs:
         http://docs.developer.amazonservices.com/en_US/subscriptions/Subscriptions_SendTestNotificationToDestination.html
+
+        destination_list ={ "Key": "sqsQueueUrl",
+         "Value": "destination" }
         """
-        raise NotImplementedError
+        if attribute_list is None:
+            raise ValueError("Attribute_list cannot be None")
+
+        data = {"Action": "SendTestNotificationToDestination",
+                "MarketplaceId": marketplace_id,
+                "Destination.DeliveryChannel": delivery_channel}
+        data.update(utils.enumerate_keyed_param("Destination.AttributeList.member", attribute_list))
+
+        return self.make_request(data, method="POST")
 
-    def create_subscription(self):
+    def create_subscription(self, marketplace_id, delivery_channel="SQS", attribute_list=None,
+                            _type=None, is_enabled=True):
```

Comment:
Recommend changing `_type` to `notification_type`, which is closer to its MWS interpretation as `NotificationType`.

Reply:
changed